src/main/java/frc/robot/Match.java

```diff
@@ -20,17 +24,48 @@
 
 public class Match extends TimedRobot
 {
+  private ADXRS450_Gyro gyro;
+  private ADXL362 accel;
+  private LinearFilter xAccelFilter;
+  private double previousXAccel;
+  private double previousVelocity;
+  private double totalDistance;
+  private double posX;
+  private double posY;
+  
   private Robot robot;
   private SendableChooser<CommandBase> autoMode;
   private CommandBase auto; 
 
+  //  ****ALL OF THESE CONSTANTS NEED TO BE TUNED BY TESTING****
+  private final static int ACCEL_SAMPLES = 5; // Num of samples of accelration in the moving average
+  private final static double ACCEL_DEADZONE = 0.0000001; // Minimum accel for it to count in the program
+  private final static double VELOCITY_DEADZONE = 0.0000001; // Minimum velocity for it to count in the program
+  private final static double STARTING_HUB_DIST = 1; // in metres
+  private final static double ACCEL_TO_CENTRE_DIST = 10000; //Distance from accelerometer to centre of rotation of robot
+  private final static double ACCEL_ANGLE = 10000; //Angle between the accelerometer and straight backwards on the robot in degrees
+  private final static double COS_OF_ACCEL_ANGLE = Math.cos(Math.toRadians(ACCEL_ANGLE));
+
   /**
    * This function is run when the robot is first started up and should be used for any
    * initialization code.
    */
   @Override
   public void robotInit()
   {
+    // instantiate the sensors
+    gyro = new ADXRS450_Gyro();
+    gyro.reset();
+    gyro.calibrate();
+
+    accel = new ADXL362(Accelerometer.Range.k8G); // Measure in the range -8g to +8g
```

Comment:
Whoa do we need 8Gs?  That seems like a lot.  Unless the robot actually takes off really fast?

Reply:
Just wondering if using lower Gs will give you more granularity in the readings.

Reply:
If 8 seems like a lot then I'll move it down to 4. Once I test how many Gs the robot has normally, I can change it to be close to the max.

Reply:
Sounds good